### ISSUE TITLE
Measure cudnn_conv_algo_search for the other conv-heavy CUDA sessions

### ISSUE THREAD

**Issue by christopherthompson81** [christopherthompson81/vernacula#191]
#190 found that ORT's CUDA EP defaults `cudnn_conv_algo_search` to `EXHAUSTIVE`, which re-benchmarks a conv the first time it sees each input **shape** — and its algo cache does not retain every shape, so it re-tunes on every shape *change*.

Measured on Kokoro (ORT 1.29, RTX 3090), same call count, only whether the shape varies between calls:

| | repeat 1 shape | cycle 8 shapes | penalty for variety |
|---|---|---|---|
| EXHAUSTIVE | 434 ms | 1271 ms | **2.93×** |
| DEFAULT | 568 ms | 789 ms | 1.39× |

`OrtSessionBuilder.AppendCuda` now takes a per-caller `cudnnConvAlgoSearch`, and **only Kokoro opts in**. Every other CUDA session in the app still inherits `EXHAUSTIVE`.

Any model whose input shape varies per call is likely paying the same penalty. Candidates, most conv-heavy first:

- [ ] the iSTFTNet vocoder (`Vocoder.cs`)
- [ ] DiariZen (`DiariZenDiarizer.cs`)
- [ ] Sortformer (`Sortformer.cs`)
- [ ] ASR encoders — Parakeet, IndicConformer, Whisper/Qwen paths
- [ ] WeSpeaker embedder, VoxLingua LID

**How to check one**, without a rebuild — `VERNACULA_ORT_CONV_ALGO` overrides the setting for every session:

```
VERNACULA_ORT_CONV_ALGO=DEFAULT     <run the workload>
VERNACULA_ORT_CONV_ALGO=EXHAUSTIVE  <run the workload>
```

Two cautions learned the hard way in #190:

- ⚠ **Benchmark with varying shapes.** A harness that repeats one utterance (as `tests/KokoroPerf` does) measures the one case where EXHAUSTIVE amortizes, and will report the opposite of the real-workload answer. Compare first-call/cold-shape numbers too.
- ⚠ **`DEFAULT` makes ORT log `Conv(...) running in Fallback mode. May be extremely slow.`** For Kokoro that path is measurably the *faster* one; the warning is not a defect signal. Don't let it scare you off, and don't assume it either — measure.

Correctness for Kokoro was unchanged (log-spec at the model's own nondeterminism floor, indistinguishable on an A/B listen). Verify per model rather than assuming; a model with an iterative loop (OmniVoice diffusion) could compound small differences the way TF32 does there.

Also outstanding from #190, smaller:

- [ ] `SegmentedSynthesis` is internal to `Vernacula.Avalonia` with no test project, so its batching driver is covered only by an end-to-end probe. Either add a test project or `InternalsVisibleTo`.
- [ ] Regenerate and redistribute the shipped Kokoro model artifact as `kokoro_batched.onnx` (`export_kokoro.py --batched`), and update the manifest.

Background: `docs/kokoro_onnx_investigation.md` Runs 24–25 (cuDNN) and 20–37 (batching).

**Comment by christopherthompson81** [comment]
Measured in #192. Short version: **"turn it on everywhere" would have been a bad idea.**

RTX 3090 / ORT 1.29, cycling five realistic input sizes per model so the shape actually changes:

| model | gain from DEFAULT | verdict |
|---|---|---|
| diarizen wespeaker | **3.09×** | DEFAULT |
| chatterbox cfm_estimator | 2.17× | DEFAULT |
| voxlingua107 LID | **1.75×** | DEFAULT |
| chatterbox mel2wav (vocoder) | 1.39× | DEFAULT |
| chatterbox flow_encoder | 1.17× | DEFAULT |
| diarizen segmentation | 0.81× | EXHAUSTIVE |
| sortformer | 0.59× | EXHAUSTIVE |
| indicconformer encoder | 0.55× | EXHAUSTIVE |
| parakeet encoder | **0.30×** | EXHAUSTIVE |

The split is architectural: conformer ASR encoders want EXHAUSTIVE (Parakeet is **3.3× slower** without it), while plain CNN stacks, vocoders and speaker-embedding nets want DEFAULT.

**Done:**

- [x] WeSpeaker embedder → DEFAULT (3.09×; embedding cosine 0.999996)
- [x] VoxLingua LID → DEFAULT (1.75×; logits cosine 1.000000000)
- [x] DiariZen segmentation — measured, keeps EXHAUSTIVE
- [x] Sortformer — measured, keeps EXHAUSTIVE
- [x] Parakeet / IndicConformer encoders — measured, keep EXHAUSTIVE
- [x] `SegmentedSynthesis` tests — no infrastructure needed; `Vernacula.Avalonia` already grants `InternalsVisibleTo` to `AsrBackendCoverage`, so the claim in this issue was wrong. Four tests added.
- [x] Model artifact — done earlier: `kokoro_batched.onnx` published, and it is now the only graph in the repo.

**Still open, and worth its own issue rather than this one:**

- ⚠ **The Chatterbox vocoder path.** `cfm_estimator` shows 2.17× and `mel2wav` 1.39×, but those are the *fallback* split graphs — the path that normally runs is the merged `conditional_decoder_loop.onnx`, which was not measured. More importantly the CFM estimator is an **iterative** solver called ~32 times per synthesis, so a per-step numerical difference compounds. That is exactly the failure mode TF32 produced in OmniVoice (`use_tf32=0` exists for that reason). A single-step check says nothing about the integrated result — this needs a real synthesis and a listen, not a microbenchmark.

Two cautions for anyone picking that up, both learned the hard way here:

- **Benchmark with varying shapes.** Timing one repeated input measures the single case where EXHAUSTIVE amortizes and reports the opposite answer.
- **Do not trust generic shape inference.** The first sweep here filled every `length` input with 0 and gave VoxLingua 40 ms of audio; `flow_encoder` read 0.87× under those shapes and 1.17× under real ones.

Also note `DEFAULT` makes ORT log `Conv(...) running in Fallback mode. May be extremely slow.` — for several of these graphs that path is measurably the faster one. Don't read the warning as a verdict either way; measure.

**Comment by christopherthompson81** [comment]
Done in #192 (merged as 3f952c1).

The measurement across every candidate session is in `docs/investigations/cudnn_conv_algo_investigation.md`. WeSpeaker (3.09×) and VoxLingua (1.75×) now opt in to DEFAULT; Parakeet, IndicConformer, Sortformer and DiariZen segmentation were measured and correctly keep EXHAUSTIVE — Parakeet is 3.3× *slower* on DEFAULT, so a global flip would have been a serious regression.

The `SegmentedSynthesis` test item turned out to need no infrastructure: `Vernacula.Avalonia` already grants `InternalsVisibleTo` to `AsrBackendCoverage`, so that claim in this issue was wrong. Four tests now pin the batching driver.

The model-artifact item was done earlier — `kokoro_batched.onnx` is published and is now the only graph in the repo.

The one genuinely open piece, the Chatterbox vocoder, is split out as #193 rather than left here: its 2.17× is on the *fallback* split graphs, and the estimator is an iterative solver where per-step error compounds, so it needs the merged loop graph measured and the audio verified — not a microbenchmark.
